jetstack/cert-manager
# Add check-spelling tool

#### What this PR does / why we need it
This adds a spell check action for pushes (to any repository with actions enabled) and for pull requests (again if the repository has actions enabled)

#3731 was merged a bit over a month ago and there have been at least ten new words misspelled in that interval.

#### Which issue this PR fixes
 *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #3858

#### Special notes for your reviewer
The easiest way to test this is:
1. fork this repository
2. enable actions on your fork
3. add the spell-check branch
4. set the spell-check branch as your default branch
5. create branches from that branch
6. create PRs into your fork targeting your default branch (=spell-check)

#### References
* The `advice.md` file is to help local tools that understand that files that end in `.md` are markdown. At some point I will probably change the tool to favor `advice.md` as the name of this input, but right now it looks for a file called `advice.txt`
* There are a couple of `README.md` files that try to help people understand how things work (in addition to the `advice.md` file
* [📹 Presentation about check-spelling](https://youtu.be/KJ2Hx-c771E?t=308) & [🎴 slides](https://docs.google.com/presentation/d/13U8a9ibqp7B1UrE8HWk7PsnZLImIwgp06vopLdmcX-s/)
* [📘 Configuration documentation](https://github.com/check-spelling/check-spelling/wiki/Configuration)
* [📐Template repository](https://github.com/check-spelling/spell-check-this/)
* [🗺️ Possible features](https://github.com/check-spelling/check-spelling/wiki/Possible-features)

#### Development process for this PR

Note that there's a lot hidden behind each of these links, the links are generally to the comment at the tip of a push, but the changes are often to the commits beneath it, so if you're interested in the details, you may want to use /compare/.... or at least /commits/... or click the `parent commit` link at times.

* [action runs](https://github.com/check-spelling/cert-manager/actions)
* [initial rebase of long running branch](https://github.com/check-spelling/cert-manager/commit/d668edb61ac2acb2547ae8c86129dbfec48732b0#commitcomment-49202667)
* [turning on experimental features](https://github.com/check-spelling/cert-manager/commit/521ed13f9b84fddb35a0464505e34e3227e93f19#commitcomment-49202637) -- this includes automatically detecting files that aren't worth checking (heuristically, any file with more unknown words than known words)
* [folding in the ignores from above, fixing the misspelled words, and turning on more prerelease bits](https://github.com/check-spelling/cert-manager/commit/7c462b1561d87cc1c084b47e2a455811844f1bb6#commitcomment-49203199)
* [updated patterns, and pull in spell-check-this template content](https://github.com/check-spelling/cert-manager/commit/a22c87b9970c8fcb4826dfeaf35d0cce7bb6858f#commitcomment-49203854) to ignore `v[0-9]`
* [prepare a branch to talk to the bot](https://github.com/check-spelling/cert-manager/commit/793514020a62cc4b989ceda19c34d6bc7d2b8472) (experimental feature)
* [PR for talking to the bot](https://github.com/check-spelling/cert-manager/pull/3) (experimental feature)
* [setting PR base](https://github.com/check-spelling/cert-manager/commit/e677f2bac9a56533fca7c975bc9027f2a7748149)
* [setting PR head](https://github.com/check-spelling/cert-manager/commit/32fdf96e95edde569ef01852d6793ef566ad7629)
* [comment from bot based on the previous two commits](https://github.com/check-spelling/cert-manager/pull/3#issuecomment-814808647)
* <details><summary>talking to the bot</summary>
  
  [comment](https://github.com/check-spelling/cert-manager/pull/3#issuecomment-814808912)
  ![animation of talking to the bot](https://raw.githubusercontent.com/check-spelling-bot/videos/quote-code/quotecode.gif)
  </details>
* [bot action handling](https://github.com/check-spelling/cert-manager/actions/runs/725719576)
* [bot submitted commit](https://github.com/check-spelling/cert-manager/pull/3/commits/651d1165da417f9f72f57c93f141185e4048df78) -- this feature requires experimental flags and only works if the default branch has the workflow enabled (and practically it requires the base branch to also have the flag enabled, otherwise there won't be an easily quotable comment to use -- if base=head, this just works)
* [rebased](https://github.com/check-spelling/cert-manager/commits/c8cf6d5e245257eb2516ad3e2c3c7bc48d7996cc)  to integrate the bot's [changes](https://github.com/check-spelling/cert-manager/blob/095ccfec8e31d25d053bfc7cddf4dbd82a722f8d/.github/actions/spelling/expect.txt)
* [dropping experimental features](https://github.com/check-spelling/cert-manager/commit/095ccfec8e31d25d053bfc7cddf4dbd82a722f8d#commitcomment-49204811) -- you can see that the latest version of the bot ignores 2 character `words` entirely, but the current shipping version still cares about them. -- this PR is making me even more confident that I want to ignore 2 character "words" (which my prerelease versions do, so 0.0.18-alpha will include that) -- the bot will automatically suggest removing obsolete items (you can see that at some of the stages here)
* [two character words added to expect](https://github.com/jsoref/cert-manager/commit/1aa82168ea7b8091e8b6597efd293339a3846506)

#### Plan

1. The current proposal is to try running with this for about a week.
  - I'll watch all PRs to the repository and help out. I can be `@`ed in GitHub (@jsoref) or on Slack (@jsoref) if people need help.
2. After that period, the team can talk about how they feel



#### Release note

```release-note
NONE
```
